doc/content/source/bcs/MFEMBoundaryCondition.md

```diff
@@ -0,0 +1,23 @@
+# MFEMBoundaryCondition
+
+## Summary
+
+Base class for MFEM boundary condition objects.
+
+## Overview
+
+`MFEMBoundaryCondition` represent boundary conditions applied to the MFEM problem, applied on the
+user-provided list of boundaries.
+
+Similar to `MFEMKernel`, the boundary condition is applied to the weak form equation that is
+labelled according to the test variable name returned from `getTestVariableName()`.
+
+`MFEMBoundaryCondition` objects come in two varieties, depending on which child class they inherit
+from:
+
+- Essential boundary conditions, inheriting from `MFEMEssentialBC`, for the application of
+  Dirichlet-like BCs removing degrees of freedom from the problem at the boundary;
+
+- Integrated boundary conditions, inheriting from `MFEMIntegratedBC`, which apply one or more
+  boundary integrators to the weak form on the specified boundary surfaces.
```

Comment:
"boundary surfaces" or just "boundaries"? Do we support 2d problems?

Reply:
"boundaries" is preferred - we don't yet, but there's no reason why we wouldn't in future

Reply:
```suggestion
  boundary integrators to the weak form on the specified boundaries.
```